node/src/chain/acceptor.rs

```diff
@@ -173,16 +174,36 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
     pub async fn spawn_task(&self) {
         let provisioners_list = self.provisioners_list.read().await.clone();
         let base_timeouts = self.adjust_round_base_timeouts().await;
+        let tip = self.tip.read().await.inner().clone();
+
+        let tip_block_voters =
+            self.get_att_voters(provisioners_list.prev(), &tip).await;
 
         self.task.write().await.spawn(
-            self.tip.read().await.inner(),
+            &tip,
             provisioners_list,
             &self.db,
             &self.vm,
             base_timeouts,
+            tip_block_voters,
         );
     }
 
+    async fn get_att_voters(
+        &self,
+        provisioners_list: &Provisioners,
+        tip: &Block,
+    ) -> Vec<VoterWithCredits> {
+        if tip.header().height == 0 {
+            return vec![];
+        };
+
+        let prev_seed = self.get_prev_block_seed().await.expect("valid seed");
+        Validator::<DB>::get_voters(tip.header(), provisioners_list, prev_seed)
+            .await
+            .expect("valid voters")
```

Comment:
Is this safe or it should return an error?

Reply:
It's unrecoverable error. If we cannot retrieve voters list we cannot proceed anyway.

Reply:
Gotcha.
In that case I would change the `expect` statement to be more clear to debug if happen and write some code comments

Otherwise we'll only have something like
```log
thread 'main' panicked at acceptor.rs:204:16:
valid voters
```

Reply:
@herr-seppia Feel free to suggest anything that you are happy with

Reply:
Nvm we'll address it when we review all unwrap/expect